src/dbnode/storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -180,14 +180,40 @@ func (s *peersSource) Read(
 			continue
 		}
 
-		r, err := s.readIndex(md,
-			namespace.IndexRunOptions.ShardTimeRanges,
-			instrCtx.span,
-			cache,
-			namespace.IndexRunOptions.RunOptions,
+		var (
+			opts           = namespace.IndexRunOptions.RunOptions
+			indexBlockSize = md.Options().IndexOptions().BlockSize()
+			idxOpts        = md.Options().IndexOptions()
+			r              result.IndexBootstrapResult
+			err            error
 		)
-		if err != nil {
-			return bootstrap.NamespaceResults{}, err
+		if s.shouldPersist(opts) {
```

Comment:
Should this check be run on `DataRunOptions` instead of `IndexRunOptions`, just like in `readData()`? I.e. if we don't persist data, we won't have any files to build index segments from.

Reply:
The evaluation should be the same as it's based on some inputted options/ranges. See earlier comment w/ links.

Reply:
+1